src/lib/OS/Vma.cc

```diff
@@ -1,99 +1,146 @@
 #include "simeng/OS/Vma.hh"
 
+#include <sys/stat.h>
+
+#include <cstddef>
+
 namespace simeng {
 namespace OS {
+HostFileMMap HostBackedFileMMaps::mapfd(int fd, size_t len, off_t offset) {
+  struct stat* statbuf = (struct stat*)malloc(sizeof(struct stat));
```

Comment:
Does this statBuf need to be a pointer at all? If yes, then please add a `free` call at the end of this function as this currently causes a memory leak.

Reply:
done